### ISSUE TITLE
[♻Housekeeping] Improve our NuGet package quality

### ISSUE THREAD

**Issue by pictos** [CommunityToolkit/Maui#520]
Today our NuGet package isn't great and after watching [this amazing talk from Claire](https://youtu.be/C_2BStepVKw) I saw that we can make our package even better!
Helping users to see our code from VS, and debug our code which will help us fix bugs (since the users can provide more complete information), we can prevent breaking changes, etc...  

I'll open one PR per item, to make it easy to review it and see if our CI will be happy.

- [x] Add Source-Link
- [ ] Turn-on package validation

**Comment by brminnick** [comment]
Hey Pedro! Was this completed via #521? Or do we need one more step to cover the `Turn-on package validation` task?

**Comment by pictos** [comment]
@brminnick yeah, I need to do the other task, at least need to confirm that it should be done

**Comment by vhugogarcia** [comment]
Hello @pictos ,

I'm just reviewing all issues and checking if they are still aplicable or fixed. By checking this issue, I noticed the only task missing is Turn-On Package Validations. After me watching [this amazing talk from Claire](https://youtu.be/C_2BStepVKw), I'm a little confuse. 

I could not find any .NET version conditionals, but I found platform conditionals [here](https://github.com/search?q=repo%3ACommunityToolkit%2FMaui+%23IF&type=code&p=1). Also, I found the supported frameworks are displayed already on NuGet here: https://www.nuget.org/packages/CommunityToolkit.Maui#supportedframeworks-body-tab even the iOS version is added there on the frameworks compatible which matches the conditionals set in the code for iOS 16.1+ 


So, my question is: Is there anything I can help with to turn-on validation on the package or is already enabled automatically, since this is an old issue, I prefer to ask hehehe :). 

Thank you

cc: @brminnick
